File: bindings/griffon-app/controllers/org/kordamp/griffon/examples/bindings/SampleController.java
package org.kordamp.griffon.examples.bindings;

import griffon.core.artifact.GriffonController;
import griffon.inject.MVCMember;
import griffon.metadata.ArtifactProviderFor;
import org.codehaus.griffon.runtime.core.artifact.AbstractGriffonController;

@ArtifactProviderFor(GriffonController.class)
public class SampleController extends AbstractGriffonController {
    @MVCMember private SampleModel model;

    public void add() {
        model.getMeasurements().add(model.nextMeasurement());
    }

    public void clear() {
        model.getMeasurements().clear();
    }
}

File: languages/kotlin/src/main/kotlin/sample/javafx/kotlin/SampleController.kt
package sample.javafx.kotlin

import griffon.core.artifact.GriffonController
import griffon.core.controller.ControllerAction
import griffon.inject.MVCMember
import griffon.metadata.ArtifactProviderFor
import org.codehaus.griffon.runtime.core.artifact.AbstractGriffonController
import javax.annotation.Nonnull
import javax.inject.Inject

@ArtifactProviderFor(GriffonController::class)
class SampleController : AbstractGriffonController() {
    @set:[MVCMember Nonnull]
    lateinit var model: SampleModel

    @Inject
    lateinit var sampleService: SampleService

    @ControllerAction
    fun sayHello() {
        model.output = sampleService.sayHello(model.input)
    }
}